x-pack/test/apm_api_integration/tests/alerts/preview_chart_error_count.spec.ts

```diff
@@ -282,4 +296,233 @@ export default function ApiTest({ getService }: FtrProviderContext) {
       });
     });
   });
+
+  registry.when(`with data loaded and using KQL filter`, { config: 'basic', archives: [] }, () => {
+    describe('error_count', () => {
+      before(async () => {
+        await generateErrorData({ serviceName: 'synth-go', start, end, synthtraceEsClient });
+        await generateErrorData({ serviceName: 'synth-java', start, end, synthtraceEsClient });
+      });
+
+      after(() => synthtraceEsClient.clean());
+
+      it('with data', async () => {
+        const options = getOptionsWithFilterQuery();
+
+        const response = await apmApiClient.readUser({
+          endpoint: 'GET /internal/apm/rule_types/error_count/chart_preview',
+          ...options,
+        });
+
+        expect(response.status).to.be(200);
+        expect(
+          response.body.errorCountChartPreview.series.some((item: PreviewChartResponseItem) =>
+            item.data.some((coordinate) => coordinate.x && coordinate.y)
+          )
+        ).to.equal(true);
+      });
+
+      it('with error grouping key in filter query', async () => {
+        const options = {
+          params: {
+            query: {
+              ...getOptionsWithFilterQuery().params.query,
+              kqlFilter:
+                'service.name: synth-go and error.grouping_key: 98b75903135eac35ad42419bd3b45cf8b4270c61cbd0ede0f7e8c8a9ac9fdb03',
```

Comment:
There is a helper `getErrorGroupingKey` that produces the error grouping key if given the correct message:

```suggestion
                `service.name: synth-go and error.grouping_key: ${getErrorGroupingKey('Error 1')}`,
```

I think it's slightly better to rely on the error messages, as they are easier to read

Reply:
updated